pandas/core/groupby.py

```diff
@@ -929,6 +929,10 @@ def mean(self):
         Compute mean of groups, excluding missing values
 
         For multiple groupings, the result index will be a MultiIndex
+
+        Do not call this function via the numpy library. numpy's mean
```

Comment:
The reason was simple: we don't want people calling these functions via the `numpy` library, so if people ever try to, the documentation is there to tell them otherwise.  Why have ambiguity by not saying anything about it and leaving the signature incompatible?